### DISCUSSION TITLE
[Question] Sink fluss,the checkpoint execution is too slow

### DISCUSSION THREAD

**Discussion by GangYang-HX** [apache/fluss#216] [Q&A]
### Search before asking

- [X] I searched in the [issues](https://github.com/alibaba/fluss/issues) and found nothing similar.

### Motivation

Deploying method: Deploying Distributed Cluster
<img width="1144" alt="image" src="https://github.com/user-attachments/assets/87838e67-774a-4900-9a7c-5147bfe754bb">
<img width="1131" alt="image" src="https://github.com/user-attachments/assets/5ff671d2-8ee9-434a-a278-f0e512390084">
The write traffic is small，but the sink operator has not yet trigger checkpoint

### Willingness to contribute

- [ ] I'm willing to submit a PR!

**Comment by GangYang-HX** [comment]
There are two key pieces of information for reference。
1.There are a lot of warn logs:"Get error write response on table bucket"
<img width="1053" alt="image" src="https://github.com/user-attachments/assets/2404879c-16b6-4064-9b4d-9a171ae29923">

2.com.alibaba.fluss.client.write.RecordAccumulator.awaitFlushCompletion Execution time is too long.
<img width="1041" alt="image" src="https://github.com/user-attachments/assets/ddc687d8-04b0-42f9-9812-0f6fe2438f09">

**Comment by swuferhong** [comment]
Hi @GangYang-HX, did you start this job immediately after you create the fluss table?  I suspect the possible cause is that you started writing data immediately after creating the table. At this point, the bucket on the tabletServer is not yet ready, so the client will retry, and it's reasonable for a warning to be logged at this time.

**Comment by GangYang-HX** [comment]
> Hi @GangYang-HX, did you start this job immediately after you create the fluss table? I suspect the possible cause is that you started writing data immediately after creating the table. At this point, the bucket on the tabletServer is not yet ready, so the client will retry, and it's reasonable for a warning to be logged at this time.

@swuferhong Hi,Thanks for your reply
I didn't pay attention to the time between creating the table and starting the task, but it should have taken a few minutes.
As of today, this warn log still exists, and it has been running for more than 18 hours.
<img width="1116" alt="image" src="https://github.com/user-attachments/assets/8836e8be-1726-4b32-8ed4-bb50275435db">

**Comment by GangYang-HX** [comment]
@swuferhong Hi,how long after creating the table is it appropriate to start the task?

**Comment by swuferhong** [comment]
> @swuferhong Hi,how long after creating the table is it appropriate to start the task?
It's strange that this problem is still occurring after such a long time theoretically. Unless there is a bug in the code, could you give me a rough description of your table creation ddl and the Flink SQL? I would like to try to reproduce it locally.

**Comment by GangYang-HX** [comment]
> > @swuferhong Hi,how long after creating the table is it appropriate to start the task?
> > It's strange that this problem is still occurring after such a long time theoretically. Unless there is a bug in the code, could you give me a rough description of your table creation ddl and the Flink SQL? I would like to try to reproduce it locally.

**Comment by GangYang-HX** [comment]
@swuferhong 
Create table script：
```
CREATE TABLE test_fluss_tb_02 (
  device_id VARCHAR,
  device_name varchar,
  version VARCHAR,
  meta_id int,
  response_id VARCHAR,
  ads_id bigint,
  client_time bigint,
  server_time  bigint,
  PRIMARY KEY (device_id) NOT ENFORCED
) WITH (
  'bucket.num' = '4'
);
```
insert script:
```
INSERT INTO `fluss_catalog`.`test_db`.`test_fluss_tb_02`
(SELECT `device_id`, `device_name`, `version`, `meta_id`, `response_id`, `ads_id`, `client_time`, `server_time`
FROM `src_xdcs`)
```
